### PR TITLE
Move cirq/ops/moment.py to cirq/circuits/moment.py

### PR DESCRIPTION
**Pull request by pavoljuhas**
Deprecate module cirq.ops.moment and attribute cirq.ops.Moment.

This finalizes #4838 by Tanuj Khattar.
Fixes https://github.com/quantumlib/Cirq/issues/4665

### CODE REVIEW

**Comment by pavoljuhas** [comment]
@tanujkhattar - ready for review.

File: cirq-core/cirq/circuits/moment_test.py [hunk 1]
```diff
@@ -305,6 +323,16 @@ def test_with_key_path():
...
 
 
+def test_with_key_path_prefix():
+    a, b, c = cirq.LineQubit.range(3)
+    m = cirq.Moment(cirq.measure(a, key='m1'), cirq.measure(b, key='m2'), cirq.X(c))
+    mb = cirq.with_key_path(m, ('b',))
+    mab = cirq.with_key_path_prefix(mb, ('a',))
+    assert mab.operations[0] == cirq.measure(a, key=cirq.MeasurementKey.parse_serialized('a:b:m1'))
+    assert mab.operations[1] == cirq.measure(b, key=cirq.MeasurementKey.parse_serialized('a:b:m2'))
+    assert mab.operations[2] is m.operations[2]
```
Comment (tanujkhattar) [inline cirq-core/cirq/circuits/moment_test.py:333]: Is this newly added test related to the change in this PR? Why do we need this?

File: cirq-core/cirq/protocols/json_test_data/PauliMeasurementGate.repr_inward [hunk 2]
```diff
@@ -2,4 +2,3 @@
     cirq.PauliMeasurementGate((cirq.X, cirq.Y, cirq.Z), cirq.MeasurementKey(name='key')),
     cirq.PauliMeasurementGate((cirq.X, cirq.Y, cirq.Z), cirq.MeasurementKey(path=('p', 'q'), name='key')),
 ]
-
```
Comment (tanujkhattar) [inline cirq-core/cirq/protocols/json_test_data/PauliMeasurementGate.repr_inward:5]: nit: unrelated change.

**Review by tanujkhattar — APPROVED** [review]
LGTM % nits. 

Thanks a lot for doing this!

File: cirq-core/cirq/circuits/moment_test.py (hunk 1 above)
Comment (pavoljuhas) [reply cirq-core/cirq/circuits/moment_test.py:333]: This fixes CI check for a test coverage of the changed code.  Because moment.py moved, the entire file appears as a new code and the CI found untested `_with_key_path_prefix_` function.

On another look at the test, it can be done without off-topic `cirq.with_key_path` so I am going to change that.

File: cirq-core/cirq/protocols/json_test_data/PauliMeasurementGate.repr_inward (hunk 2 above)
Comment (pavoljuhas) [reply cirq-core/cirq/protocols/json_test_data/PauliMeasurementGate.repr_inward:5]: This was from my commit hook which balks at trailing whitespace.  Will put it back as it were.

File: cirq-core/cirq/circuits/moment_test.py (hunk 1 above)
Comment (pavoljuhas) [reply cirq-core/cirq/circuits/moment_test.py]: done - dbf820048b779d90f3dab8d2e15f4410effb85b3
